src/Microsoft.CodeAnalysis.Razor.Workspaces/SourceTextExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.CodeAnalysis.Text
+{
+    internal static class SourceTextExtensions
+    {
+        public static RazorSourceDocument GetRazorSourceDocument(this SourceText sourceText, string fileName)
```

Comment:
This is debatable. I couldn't add a `RazorSourceDocument.Create(SourceText)` overload because we can't take a dependency on CodeAnalysis.Text from Razor.Language and C# doesn't allow static extension methods. So I added this extension method instead. You can argue this doesn't provide much value. I could be convinced either way.

Reply:
I think this is fine. It's internal to our usage.